include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -122,13 +103,14 @@ class bi_fm_index
      * \{
      */
     //!\brief The type of the bidirectional cursor.
-    using cursor_type = bi_fm_index_cursor<bi_fm_index<is_collection, sdsl_index_type>>;
-    //!\brief The type of the unidirectional cursor on the original text.
-    using fwd_cursor_type = fm_index_cursor<fm_index_type>;
-    //!\brief The type of the unidirectional cursor on the reversed text.
-    using rev_cursor_type = fm_index_cursor<rev_fm_index_type>;
+    using cursor_type = bi_fm_index_cursor<bi_fm_index<alphabet_t, text_layout_mode, sdsl_index_type>>;
+    //!\brief The type of the unidirectional cursor.
+    using uni_cursor_type = fm_index_cursor<fm_index_type>;
     //!\}
 
+    //!\brief Indicates whether index is built over a collection.
+    static constexpr bool is_collection = text_layout_mode;
```

Comment:
Can we instead of exposing `is_collection` change it to`text_layout_mode`?

We have to change all `is_collection` to `text_layout_mode = text_layout::collection`?

Reply:
> Can we instead of exposing `is_collection` change it to`text_layout_mode`?

We could.
> 
> We have to change all `is_collection` to `text_layout_mode = text_layout::collection`?
Yes. That's why I didn't. But maybe I can `static_cast` it to `bool`?